File: practice/algorithms/implementation/jumping_on_the_clouds/JumpingOnTheClouds.js
'use strict';

const processData = input => {
    const lines = input.split('\n');
    const clouds = lines[1].split(' ').map(i => parseInt(i));
    let steps = 0;
    let  index = 0;
    while (index < clouds.length - 1) {
        if((index + 2 < clouds.length) && clouds[index + 2] != 1) {
            index = index + 2;
        } else {
            index = index + 1;
        }
        steps += 1;
    }
    console.log(steps);
};

process.stdin.resume();
process.stdin.setEncoding("ascii");

let _input = "";
process.stdin.on("data", input => _input += input);
process.stdin.on("end", () => processData(_input));

File: practice/algorithms/implementation/jumping_on_the_clouds/JumpingOnTheClouds.java
import java.util.*;

public class JumpingOnTheClouds {

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        int n = Integer.parseInt(stdin.nextLine());
        String[] numbers = stdin.nextLine().split(" ");
        int[] clouds = new int[n];
        for(int i = 0; i < n; i++) {
            clouds[i] = Integer.parseInt(numbers[i]);
        }

        int steps = 0;
        int index = 0;
        while (index < clouds.length - 1) {
            if((index + 2 < clouds.length) && clouds[index + 2] != 1) {
                index = index + 2;
            } else {
                index = index + 1;
            }
            steps += 1;
        }
        System.out.println(steps);
        stdin.close();
    }
}